### PR TITLE
feat: Added the login and register endpoints

### PR DESCRIPTION
**Pull request by frederikpyt**
## Description
1. Restructured the folder structure for better overview
2. Added the login and register endpoints
3. Installed fastify-zod and zod for better request body validation
4. Installed and setup swagger to get a better overview over our endpoints

## Type of Change

- [x] ✨ New feature (non-breaking change which adds functionality)
- [ ] 🛠️ Bug fix (non-breaking change which fixes an issue)
- [ ] ❌ Breaking change (fix or feature that would cause existing functionality to change)
- [x] 🧹 Code refactor
- [ ] ✅ Build configuration change
- [x] 📝 Documentation
- [ ] 🗑️ Chore

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,3 +1,6 @@
+## API documentation (Swagger)
+Open [http://localhost:3000/api/docs](http://localhost:3000/api/docs) to view it in the browser.
+
 ## Available Scripts
```
Comment (kristianbinau) [inline README.md:4]: How to run tests and setup test db?

File: README.md [hunk 2]
```diff
@@ -1,3 +1,6 @@
+## API documentation (Swagger)
+Open [http://localhost:3000/api/docs](http://localhost:3000/api/docs) to view it in the browser.
```
Comment (kristianbinau) [inline README.md:2]: Link wrong

Comment (frederikpyt) [reply README.md:2]: Wrong if you use our nginx proxy, but correct if not

File: src/modules/auth/auth.schema.ts [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+const createUserSchema = z.object({
+    ...userCore,
+    password: z.string({
+        required_error: "Password is required",
+        invalid_type_error: "Password must be a string",
+    }),
+});
+
+const createUserResponseSchema = z.object({
+    id: z.number(),
```
Comment (kristianbinau) [inline src/modules/auth/auth.schema.ts:24]: Dont send id back : )

File: src/modules/auth/auth.schema.ts [hunk 4]
```diff
@@ -0,0 +1,49 @@
+import { z } from "zod";
```
Comment (kristianbinau) [inline src/modules/auth/auth.schema.ts:1]: This is so cool

File: README.md (hunk 1 above)
Comment (frederikpyt) [reply README.md:4]: Run tests: https://github.com/HF5-PROJECT/api/blob/33feb9d5ffa3fd192173d50e9db1c44f5938e09d/README.md?plain=1#L17-L19

Setup DB:
https://github.com/HF5-PROJECT/api/blob/33feb9d5ffa3fd192173d50e9db1c44f5938e09d/README.md?plain=1#L25-L27

File: src/plugins/jwt.ts [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+declare module "fastify" {
+    interface FastifyRequest {
+        jwt: JWT;
+    }
+}
+
+export default fastifyPlugin(
+    async (fastify: FastifyInstance, opts: FastifyPluginOptions) => {
+        fastify.register(fastifyJwt, {
+            secret: "supersecretsecret",
```
Comment (kristianbinau) [inline src/plugins/jwt.ts:19]: Change to env value

File: src/__test__/server.test.ts [hunk 6]
```diff
@@ -12,7 +12,7 @@ describe("GET /api/health", () => {
         await fastify.close();
     });
 
-    it("should return status OK", async () => {
+    it("should return status 200 and token", async () => {
```
Comment (kristianbinau) [inline src/__test__/server.test.ts:15]: and token? We dont send token here? Dont think we should anyways.

```suggestion
    it("should return status 200", async () => {
```

**Review by kristianbinau — CHANGES_REQUESTED** [review]
Really cool

File: src/__test__/server.test.ts (hunk 6 above)
Comment (frederikpyt) [reply src/__test__/server.test.ts:15]: Brainfart

File: src/plugins/jwt.ts (hunk 5 above)
Comment (frederikpyt) [reply src/plugins/jwt.ts:19]: Okay

**Review by kristianbinau — CHANGES_REQUESTED** [review]
Just small change

File: .env.example [hunk 7]
```diff
@@ -1,6 +1,7 @@
 # Server
 HOST=0.0.0.0
 PORT=3000
+SECRET=oækejm(=U?)Suifds8j+0fa8//AS=Y)
```
[the remark this answers is not in the document]
Comment (frederikpyt) [reply .env.example]: Done

**Review by kristianbinau — APPROVED** [review]
Awesome! 🥇

**Pull request merged**